facebook/pyre-check
# Adds AllOf constraint to Pysa DSL

Adds support for a new constraint - `AllOf` to Pysa DSL, to be used in
the where clause. This query would find methods which obey all of the
conditions specified inside it and then model only such methods.

Screenshot:
<img width="1440" alt="Screenshot 2021-11-03 at 8 23 44 PM" src="https://user-images.githubusercontent.com/8947010/140084198-78821e43-5edb-4469-8f53-94a6729ea812.png">

Test Plan
- create a new model query with the new `AllOf` constraint.
- run `pyre analyse` with `dump-model-results-to` after compiling the source (from this branch)
- See the constraint in action. Maybe, play with it a bit...

Fixes: https://github.com/MLH-Fellowship/pyre-check/issues/86
Signed-off-by: Abishek V Ashok <abishekvashok@gmail.com>
